src/ray/gcs/redis_context.h

```diff
@@ -0,0 +1,63 @@
+#ifndef RAY_GCS_REDIS_CONTEXT_H
+#define RAY_GCS_REDIS_CONTEXT_H
+
+#include <functional>
+#include <memory>
+#include <unordered_map>
+
+#include "ray/id.h"
+#include "ray/status.h"
+#include "ray/util/logging.h"
+
+struct redisContext;
+struct redisAsyncContext;
+struct aeEventLoop;
+
+namespace ray {
+
+namespace gcs {
+
+class RedisCallbackManager {
+ public:
+  using RedisCallback = std::function<void(const std::string &)>;
+
+  static RedisCallbackManager &instance() {
+    static RedisCallbackManager instance;
+    return instance;
+  }
+
+  int64_t add(const RedisCallback &function);
+
+  RedisCallback &get(int64_t callback_index);
+
+ private:
+  RedisCallbackManager() : num_callbacks(0){};
```

Comment:
We should keep it simple if possible: remove `num_callbacks` and its maintenance, since it's just `callbacks_.size()`.